Website/src/rendering/src/components/Checkout/PanelComments.tsx

```diff
@@ -14,11 +21,9 @@ const PanelComments = (props: PanelCommentsProps): JSX.Element => {
             name="order-comments"
             id="order-comments"
             rows={3}
-            value={props.orderComments}
             placeholder="Leave your comment for us here..."
-            onChange={(event) => {
-              props.onEditComments(event.target.value);
-            }}
+            defaultValue={order?.Comments}
+            onBlur={(event) => updateComments(event.target.value)}
```

Comment:
If I enter a comment, stay focused on the box, then click on the "Review order" button, the calls to save the comments and to load the next page will run almost simultaneously.

Can we run into a race condition where the call that load the order details on the order-review page does not get the latest comment?

I am wondering whether we should revert to handling saving the comments when onChange?

Reply:
Thanks Neli for the research. There is another PR by Crhistian that will add improvements to the checkout page with loading states and animations.
For now we can simply put a todo: `TODO: Look to disable the "Review order" button while the comment is being saved.`

We could look at the todo when all of this will be merged together.